### ISSUE TITLE
Problem with multi-county block groups when variables are stored in a vector

### ISSUE THREAD

**Issue by walkerke** [walkerke/tidycensus#94]
This fails: 

```r
vars <- paste0("B19001_00", 2:8)

arlington_bgs <- get_acs(geography = "block group", 
                         state = "TX", 
                         county = c("Tarrant", "Dallas"), 
                         variables = vars, 
                         geometry = TRUE) 
```

But this works: 

```r
arlington_bgs <- get_acs(geography = "block group", 
                         state = "TX", 
                         county = c("Tarrant", "Dallas"), 
                         variables = paste0("B19001_00", 2:8), 
                         geometry = TRUE) 
```

Fixing this will require a re-factoring of how I handle multi-county block group calls.

**Comment by plnnr** [comment]
I think I have a similar problem with states and counties, but not variables. I made a function that returns the median household income by race for the year 2000. I set default parameters equal to vectors. The function fails when vectors are specified, but the function works if only a single state or single county is passed. The Gist file is here. https://gist.github.com/plnnr/8a218345425c13a47078a83b331d0df1 I'd love if you could take a look... ?

**Comment by walkerke** [comment]
@plnnr thanks for this, I'll take a look.  I think this will need an internal refactor to solve both of these but I think that's due.

**Comment by plnnr** [comment]
I also notice though that even the get_acs() function doesn't seem to handle a year that is coded as a variable. I made another Gist code here (https://gist.github.com/plnnr/d6809b5d2c9ee42401da26fe1ecfb2ec) that _should_ work, but it fails saying that the object specified in the function call ("acs_year") can't be found. Even when I hard-code it within the function, it doesn't recognize the object.

**Comment by walkerke** [comment]
Any chance you could provide a minimal reproducible example (https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example)?  That'll help me debug.  In my preliminary testing I'm not noticing problems with states/counties/years as variables.

**Comment by plnnr** [comment]
@walkerke Here is a more clear example with better comments:
https://gist.github.com/plnnr/1af71d9b7b290f93bb8612710a6d12dd

**Comment by walkerke** [comment]
This is now fixed in tidycensus 0.9.2
